core/src/nucleus/ribosome/capabilities.rs

```diff
@@ -0,0 +1,46 @@
+/// capabilities implements the capability request functionality used to check
+/// that a given capability has been granted for actions like zome calls
+use holochain_core_types::{
+    cas::content::Address,
+    signature::{Provenance, Signature},
+};
+
+/// a struct to hold the capability information needed to make any capability request,
+/// namely the provenance of the request (the agent address an signature) and the
+/// actual token being used to make the request
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, Hash)]
+pub struct CapabilityRequest {
+    pub cap_token: Address,
+    pub provenance: Provenance,
+}
```

Comment:
Inquiry: Should `CapabilityRequest` include the whole function call as well?

Context: We'll need to define a serialized representation of a signed call that we can create and sign in UIs and other clients and also deserialize and check in core. Should this become that?

Reply:
I thought about that.  In the general case the provenance should point to what it's a provenance for, and the request could be for things other than zome calls.  Because right now there is only one thing you can request you always know contextually what is signed in the provenance, so it's not needed.